zipline/finance/blotter.py

```diff
@@ -329,20 +329,23 @@ def get_transactions(self, bar_data):
                     if not order.open:
                         closed_orders.append(order)
 
+        self.prune_orders(closed_orders)
+
+        # FIXME this API doesn't feel right (returning two things here)
+        return transactions, None
+
+    def prune_orders(self, closed_orders):
         # remove all closed orders from our open_orders dict
         for order in closed_orders:
             sid = order.sid
             try:
                 sid_orders = self.open_orders[sid]
                 sid_orders.remove(order)
-            except KeyError:
+            except ValueError:
```

Comment:
I'll add a comment, but we expect this in a downstream subclass that is connected to a broker, and the broker is canceling orders.  There's some scenario there that causes us to try to remove an order that we've already removed locally (probably due to EOD cancel policy on our own end), and thus is happening.

Either way, we're trying to remove something that is already removed, so we shouldn't blow up on it.